src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -700,6 +701,11 @@ def filter_out_esm_packages(self, packages, package_versions):
             self.status_handler.add_error_to_status("{0} patches requires Ubuntu Pro for Infrastructure with Extended Security Maintenance".format(len(esm_packages)), Constants.PatchOperationErrorCodes.UA_ESM_REQUIRED) # Set the error status with the esm_package details. Will be used in portal.
             self.telemetry_writer.write_event("Filter esm packages [EsmPackagesCount={0}]".format(len(esm_packages)), Constants.TelemetryEventLevel.Informational)
 
-        self.composite_logger.log_debug("Filter esm packages : [TotalPackagesCount={0}][EsmPackagesCount={1}]".format(len(packages), len(esm_packages)))
+        # Log the Esm packages that will be patched in case of an Ubuntu Pro attached VM.
+        security_esm_update_query_success, security_esm_updates, security_esm_updates_versions = self.get_security_esm_updates()
```

Comment:
I think there should be separate function for logging ESM packages selected to install. The responsibility of filter_out_esm_packages should only be filtering out ESM required packages from the packages to be installed.

The function filter_out_esm_packages might be called from multiple places in future but we need to log ESM packages to install only once. Creating a separate function will also make code readability better.

Also, the variable names are confusing in this function like for following line esm_packages_selected_to_install contains packages from the list non_esm_packages.

esm_packages_selected_to_install = [package for package in non_esm_packages if package in security_esm_updates]

Reply:
Unfortunately No. This method is used to just abstract Pro specific changes from the common code. We will not be reusing anywhere. If not here, we will have to unnecessarily create other set of methods which will be stubbed in YUM and Zypper.

So, I will keep this logic here as of now.

Yes, Variable names seems to be confusing. will update it.

Reply:
@GAURAVRAMRAKHYANI : I have made changes to move the code change to call esm_updates to one place. Please check this.